src/types.py

```diff
@@ -162,6 +162,7 @@ class TypeData:
     array_dim: Optional[int] = None  # K_ARRAY
     struct: Optional["StructDeclaration"] = None  # K_STRUCT
     enum: Optional[Enum] = None  # K_INT
+    new_field_within: Set["StructDeclaration"] = field(default_factory=set)
```

Comment:
It turns out to be almost unnecessary given the other change I made to add_imm (which keeps us doing `var_x += 8` instead of creating a struct member and doing `var_x = (Type*) &var_x->unk8`). But there's one place in mm that still needs it, and it seems like an all-around sensible thing to have. (If we start supporting merging structs together the design will need to change a little.)